### PR TITLE
EVG-14406 continue on api errors in git.get_project

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: agent/command/git.go [hunk 1]
```diff
@@ -254,7 +254,9 @@ func (c *gitFetchProject) buildCloneCommand(ctx context.Context, conf *internal.
 			// https://docs.github.com/en/rest/guides/getting-started-with-the-git-database-api#checking-mergeability-of-pull-requests
 			commitToTest, err = c.waitForMergeableCheck(ctx, opts.owner, opts.repo, conf.GithubPatchData.PRNumber)
 			if err != nil {
-				return nil, err
+				logger.Task().Error(errors.Wrap(err, "error checking if pull request is mergeable"))
```
Comment (bsamek) [inline agent/command/git.go:257]: I don't think the agent should wrap.

**Review by bsamek — APPROVED** [review]

File: agent/command/git.go (hunk 1 above)
Comment (john-m-liu) [reply agent/command/git.go:257]: this one is to give more context on the error for the user. Currently it's just `error getting pull request data from Github: GET https://api.github.com/repos/10gen/dsi/pulls/875: 404 Not Found []`. I'd prefer to keep it

**Review by hhoke — APPROVED** [review]
LGTM (for tomorrow)

File: agent/command/git.go (hunk 1 above)
Comment (bsamek) [reply agent/command/git.go:257]: SGTM

**Comment by john-m-liu** [comment]
evergreen merge

**Pull request merged**